### PR TITLE
docker: 1.10.3 -> 1.11.1

### PR DESCRIPTION
**Pull request by offlinehacker**
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Now docker requires containerd and runc, as it was split in separated components. I still have to test this on nixos.

### CODE REVIEW

**Comment by nathan7** [comment]
Minor nit: fix the commit message typo on bd090c9236ec1f71e42d8707cdddb90b83391904?
I'll have a go at testing this on NixOS later today — I succesfully packaged 1.11.0 when it was released, but I ran into some serious breakage.

File: pkgs/applications/virtualization/containerd/default.nix [hunk 1]
```diff
@@ -14,19 +13,16 @@ stdenv.mkDerivation rec {
     sha256 = "07axhrvy45zwnsrvr2618227bzrnqdcjdl7j0mnrhvlryypiic0l";
   };
 
-  buildInputs = [ go libapparmor libseccomp ];
-
-  makeFlags = ''BUILDTAGS+=seccomp BUILDTAGS+=apparmor'';
+  buildInputs = [ makeWrapper go ];
```
Comment (fpletz) [inline pkgs/applications/virtualization/containerd/default.nix]: Is seccomp disabled after this commit? I don't think it's a good idea from a security standpoint.

**Comment by fpletz** [comment]
This doesn't build for me because the runc build won't find libseccomp.

**Comment by steveeJ** [comment]
@offlinehacker Have you made progress with this?

**Comment by nathan7** [comment]
If there is no progress on this, I have a working build I can contribute and adjust to nixpkgs style somewhere this week.

**Comment by puffnfresh** [comment]
0.12.0 is out now.

**Comment by joachifm** [comment]
Any progress on this?

**Comment by offlinehacker** [comment]
I'm closing this in favor of https://github.com/NixOS/nixpkgs/pull/18481

**Pull request closed without merging**